### PR TITLE
complete -profile devel usage string

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by ejgallego** [comment]
I have no idea what this is doing, can you add a description to the commit?

File: tools/configure/cmdArgs.ml [hunk 1]
```diff
@@ -143,6 +147,8 @@ let check_absolute = function
 let args_options = Arg.align [
   "-prefix", arg_string_option (fun p prefix -> check_absolute prefix; { p with prefix }),
     "<dir> Set installation directory to <dir> (absolute path required)";
+  "-local", arg_set (fun p -> local p),
+    " Set prefix to \""^local_prefix^"\" and disable installation";
```
Comment (ejgallego) [inline tools/configure/cmdArgs.ml:151]: For sure I don't want -local to be back, what's the need?

Comment (SkySkimmer) [reply tools/configure/cmdArgs.ml:151]: People keep asking for it

Comment (Alizter) [reply tools/configure/cmdArgs.ml:151]: https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/.2E.2Fconfigure.20-local.20no.20longer.20exists.2C.20what.20to.20use.20now.3F/near/269321868

Comment (ejgallego) [reply tools/configure/cmdArgs.ml:151]: That's not people asking for it, they were installing from sources in a weird way; they can just use a supported method actually. `-local` was all about having a non-standard layout, which goes against the direction all modern build systems take.

Comment (ejgallego) [reply tools/configure/cmdArgs.ml:151]: GNU `stow` has existed for decades for that use case, for example.

Comment (SkySkimmer) [reply tools/configure/cmdArgs.ml:151]: >That's not people asking for it, they were installing from sources in a weird way

I don't understand what you mean, the answer they got was to use profile devel, which is just local plus some other stuff. What's wrong with exposing local when we already have it as part of devel?

Comment (ejgallego) [reply tools/configure/cmdArgs.ml:151]: Actually the solution to use `-profile devel` is also wrong, that's not a release build and will not be optimized properly.

Comment (ejgallego) [reply tools/configure/cmdArgs.ml:151]: What is wrong is that you are encouraging people to use an option that IMHO is wrong for their use case.

Comment (SkySkimmer) [reply tools/configure/cmdArgs.ml:151]: How do you know their use case? And what's wrong with local?

Comment (ejgallego) [reply tools/configure/cmdArgs.ml:151]: I assumed their use case is to actually use Coq, not to develop Coq. If so, the right answer is to use a release build.

Comment (ejgallego) [reply tools/configure/cmdArgs.ml:151]: @robbertkrebbers what's your use case?

**Comment by robbertkrebbers** [comment]
My use case is to build any version of Coq from sources (that could be my own fork, some pull request, some git checkout, etc). For this I typically want an optimized build, not a version with debugging stuff enabled. I also don't want to run `make install`, but I want the binaries (and compiled stdlib etc) to be just dumped in the same folder as the sources.

I always used `-local` for this, and that worked just fine. I assume that my use case is so common there should be a proper alternative for it.

File: tools/configure/cmdArgs.ml (hunk 1 above)
Comment (Zimmi48) [reply tools/configure/cmdArgs.ml:151]: > I don't understand what you mean, the answer they got was to use profile devel, which is just local plus some other stuff. What's wrong with exposing local when we already have it as part of devel?

That's the answer you give them, but I already complained about this answer in https://coq.discourse.group/t/how-to-use-a-local-installation-of-coq/1507.

Comment (SkySkimmer) [reply tools/configure/cmdArgs.ml:151]: That's just restating that local was removed, ie solved by ths PR

**Comment by Zimmi48** [comment]
If you don't like opam, have you considered using Nix? With the additional use of the [Coq Nix Toolbox](https://github.com/coq-community/coq-nix-toolbox), all the use cases you mention are supported.

Anyway, yes, even without the `-local` option, there should be a ready-to-use, install-like layout available under `_build/default` or `_build_vo` as indicated by @ejgallego in https://coq.discourse.group/t/how-to-use-a-local-installation-of-coq/1507.

**Comment by robbertkrebbers** [comment]
[Apologies for late response]

> Anyway, yes, even without the -local option, there should be a ready-to-use, install-like layout available under _build/default or _build_vo as indicated by @ejgallego in https://coq.discourse.group/t/how-to-use-a-local-installation-of-coq/1507.

I tried this, but it does not quite seem to work:

```shell
./configure
# During configure I am asked for folders for the install, but since I don't install, should I just hint <enter>?
make
_build/install/default/bin/coqc
```

This fails with "cannot guess a path for Coq libraries; please use -coqlib option or ensure you have installed the package containing Coq's stdlib (coq-stdlib in OPAM) If you intend to use Coq without a standard library, the -boot -noinit options must be used."

**Comment by ejgallego** [comment]
You need make states right? Tho we will fully overhaul the current mess next week so I suggest you wait a few more days as all we say now will become obsolete

**Comment by robbertkrebbers** [comment]
> You need make states right?

I don't know what you mean by that.

> Tho we will fully overhaul the current mess next week so I suggest you wait a few more days as all we say now will become obsolete

It's good that there will be an overhaul, but that will be for Coq 8.16 I presume? It still would be nice to have a solution for Coq 8.15, even if only a hackish solution exists.

**Comment by ejgallego** [comment]
Coq 8.15 supports that just fine, what are the exact commands you are typing? That make call is certainly weird, the error is of course expected with it.

**Comment by ejgallego** [comment]
Yes the overhaul will be for 8.16

**Comment by robbertkrebbers** [comment]
> what are the exact commands you are typing?

In Coq 8.13 I used to type:

```
./configure -local
make
bin/coqc
```

This used to work just fine.

With Coq 8.15.1 I tried:

```
./configure
# During configure I am asked for folders for the install, but since I don't install, should I just hint <enter>?
make
_build/install/default/bin/coqc
```

But that does not work. Do you have the right commands that I should use?

**Comment by ejgallego** [comment]
> It turns out that `-prefix` with a subfolder of my Coq tree and then running `make install` works. It's a step more than `-local`, but it seems fine.

I'm glad you found a solution; indeed, not surprisingly after you learn how build system related stuff goes even in the best industry places, we botched the transition quite a bit :) hopefully soon we mostly fix the current mess.

**Comment by SkySkimmer** [comment]
Removed -local

**Comment by ppedrot** [comment]
Status of this w.r.t. 8.16?

**Comment by SkySkimmer** [comment]
waiting for assignee

**Comment by ppedrot** [comment]
I am not competent to assign this PR, and the 8.16 branch is getting closer. @ejgallego maybe?

**Comment by ppedrot** [comment]
If this is not merged quickly, it'll have to be postponed. I don't think it qualifies as a fix, so the backportability status is debatable.

**Pull request closed without merging**